python/llm_util.py

```diff
@@ -0,0 +1,153 @@
+import mgp
+from enum import Enum
+
+
+class Parameter(Enum):
+    END = "end"
+    NODE_PROPS = "node_props"
+    PROPERTY = "property"
+    RELATIONSHIPS = "relationships"
+    REL_PROPS = "rel_props"
+    START = "start"
+    TYPE = "type"
+
+
+class OutputType(Enum):
+    RAW = "raw"
+    PROMPT_READY = "prompt_ready"
+
+
+class SchemaGenerator(object):
+    def __init__(self, context, output_type):
+        self._type = output_type.lower()
+        self._node_counter = 0
+        self._all_node_properties_dict = {}
+        self._all_relationship_properties_dict = {}
+        self._all_relationships_list = []
+
+        self._generate_schema(context)
+
+    @property
+    def type(self):
+        return self._type
+
+    @property
+    def node_counter(self):
+        return self._node_counter
+
+    def get_schema(self):
+        if self._type == OutputType.RAW.value:
```

Comment:
If you will add some new enum values, I think it is better to use a switch statement as it probably should detect that we are missing some enum processing.

Reply:
Afaik switch statement is introduced with Python 3.10 and MAGE is on Python 3.9 still? Or are you talking about something else?

Reply:
Just a regular switch statement. Oh, it was introduced in 3.10, didn't know that. Then it is okay like this